codec/src/types/primitives.rs

```diff
@@ -40,6 +43,39 @@ impl_numeric!(i128, get_i128, put_i128);
 impl_numeric!(f32, get_f32, put_f32);
 impl_numeric!(f64, get_f64, put_f64);
 
+// Usize implementation
+// Since most values refer to a length or size, the type is encoded as a varint.
```

Comment:
I am guessing this is because length/size tends to be small, so you save a few bytes this way?